internal/containerizer/dockerfilecontainerizer.go

```diff
@@ -91,53 +92,190 @@ func (d *DockerfileContainerizer) GetContainer(plan plantypes.Plan, service plan
+	sourceCodeDir := service.SourceArtifacts[plantypes.SourceDirectoryArtifactType][0] // TODO: what about the other source artifacts?
 
-	// Create the Dockerfile.
-	dockerfileTemplatePath := filepath.Join(containerizerDir, "Dockerfile")
-	dockerfileTemplateBytes, err := ioutil.ReadFile(dockerfileTemplatePath)
+	relOutputPath, err := filepath.Rel(plan.Spec.Inputs.RootDir, sourceCodeDir)
 	if err != nil {
-		log.Errorf("Unable to read the Dockerfile template at path %q Error: %q", dockerfileTemplatePath, err)
+		log.Errorf("Failed to make the source code directory %q relative to the root directory %q Error: %q", sourceCodeDir, plan.Spec.Inputs.RootDir, err)
 		return container, err
 	}
-	dockerfileTemplate := string(dockerfileTemplateBytes)
-	sourceCodeDir := service.SourceArtifacts[plantypes.SourceDirectoryArtifactType][0] // TODO: what about the other source artifacts?
 
+	//　1. Execute detect to obtain the json response from the .sh file
 	output, err := d.detect(containerizerDir, sourceCodeDir)
 	if err != nil {
 		log.Errorf("Detect using Dockerfile containerizer at path %q on the source code at path %q failed. Error: %q", containerizerDir, sourceCodeDir, err)
 		return container, err
 	}
 	log.Debugf("The Dockerfile containerizer at path %q produced the following output: %q", containerizerDir, output)
 
-	dockerfileContents := dockerfileTemplate
-	if output != "" {
-		m := map[string]interface{}{}
-		if err := json.Unmarshal([]byte(output), &m); err != nil {
-			log.Errorf("Unable to unmarshal the output of the detect script at path %q Output: %q Error: %q", containerizerDir, output, err)
-			return container, err
+	// 2. Parse json output
+	if output == "" {
+		log.Warnf("Output variable is empty")
+		return container, nil
+	}
+
+	m := map[string]interface{}{}
+	if err := json.Unmarshal([]byte(output), &m); err != nil {
+		log.Errorf("Unable to unmarshal the output of the detect script at path %q Output: %q Error: %q", containerizerDir, output, err)
+		return container, err
+	}
+
+	// Final multiline string containing the generated Dockerfile will be stored here
+	dockerfileContents := ""
+	// Filled segments will be stored here
+	segmentSlice := []string{}
+
+	// 2.1 Obtain segmentRecords slice for both cases (segment-based, df)
+	segmentRecords := []map[string]interface{}{}
+
+	if _, ok := m["type"]; ok {
+		// segment-based
+		if _, segmentsFound := m["segments"]; segmentsFound {
+			if segments, ok := m["segments"].([]interface{}); ok {
+				for _, segment := range segments {
+					if rec, ok := segment.(map[string]interface{}); ok {
+						segmentRecords = append(segmentRecords, rec)
+					}
+				}
+			}
 		}
 
-		if value, ok := m[containerizerJSONPort]; ok {
-			portToExpose := int(value.(float64)) // Type assert to float64 because json numbers are floats.
+	} else {
+		// Add the fixed segment id associated to the full Dockerfile and append to segmentRecords
+		m["segment_id"] = "Dockerfile"
```

Comment:
Since we use "segment_id" in multiple places. Would be good to capture all "special" keys as consts in the start of the file.